test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,352 @@
+	"antrea.io/antrea/test/e2e/utils"
+)
+
+/*
+1) Create the KIND cluster.
+2) Once the cluster is up , create a service for custom DNS.
+3) Configure and update Antrea configmap with above service's IP.
+
+5) Create agnHost 2 pods (deployment?).
+6) Get IP of one of the pods of agnHost.
+7) create and configure the custom CoreDNS configMap with the IP received above.
+8) Create a custom DNS pod and configure it to use above configMap with agnHost pod's IP.
+9) Create and apply antrea FQDN policy.
+10) Deploy antrea-toolbox.
+
+11) curl the FQDN from within toolbox.
+12) imitate caching the IP belonging to above FQDN resolution by keeping it in a variable.
+13) change the existing IP in dns configmap with the other agnHost pod's IP.
+14) wait for new IP to get updated in configMap and let the changes be reflected in dns pod.
+15) curl the FQDN again with old IP , simulating usage of cache -- and it must fail with no connectivity.
+*/
+
+const (
+	testFullyQualifiedDomainName = "fqdn-test-pod.lfx.test"
+	customDnsConfigName          = "custom-dns-config"
+	customDnsPodName             = "custom-dns-server"
+
+	customDnsLabelKey   = "app"
+	customDnsLabelValue = "custom-dns"
+
+	fqdnPodSelectorLabelKey   = "app"
+	fqdnPodSelectorLabelValue = "fqdn-cache-test"
+)
+
+func TestFQDNPolicyWithCachedDNS(t *testing.T) {
+	const (
+		agnHostLabelKey      = "app"
+		agnHostLabelValue    = "agnhost"
+		agnHostPodNamePreFix = "agnhost-"
+
+		dnsPort              = 53
+		customDnsServiceName = "custom-dns-service"
+	)
+	skipIfAntreaPolicyDisabled(t)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	customDnsService, err := data.CreateServiceWithAnnotations(customDnsServiceName, data.testNamespace, dnsPort, dnsPort, corev1.ProtocolUDP,
+		map[string]string{customDnsLabelKey: customDnsLabelValue}, false, false, corev1.ServiceTypeClusterIP, ptr.To[corev1.IPFamily](corev1.IPv4Protocol), map[string]string{})
+	require.NoError(t, err, "Error when creating custom DNS service: %v", err)
+
+	setDnsServerAddressInAntrea(t, data, customDnsService.Spec.ClusterIP)
+	defer setDnsServerAddressInAntrea(t, data, "")
+
+	podCount := 2
+	agnHostPodIps := make([]*PodIPs, podCount)
+	for i := 0; i < podCount; i++ {
+		agnHostPodIps[i] = createHttpAgnhostPod(t, data, randName(agnHostPodNamePreFix), map[string]string{agnHostLabelKey: agnHostLabelValue})
+	}
+
+	ipv4, ipv6 := extractIPs(agnHostPodIps[0])
+
+	dnsConfigData := createDnsConfig(t, ipv4, ipv6, testFullyQualifiedDomainName)
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      customDnsConfigName,
+			Namespace: data.testNamespace,
+		},
+		Data: dnsConfigData,
+	}
+	customDnsConfigMapObject, err := data.CreateConfigMap(configMap)
+	require.NoError(t, err, "failed to create custom dns ConfigMap: %v", err)
+
+	createDnsPod(t, data)
+
+	k8sUtils, err = NewKubernetesUtils(data)
```

Comment:
There is no need for `NewKubernetesUtils` here. `k8sUtils` is not used anywhere. The call to `k8sUtils.runDNSQuery` is misleading as it should be `data.runDNSQuery`.